src/webgpu/api/operation/render_pass/storeOp.spec.ts

```diff
@@ -1,23 +1,352 @@
+      colorAttachments.push(
+        t.device.createTexture({
+          format: kColorFormat,
+          size: { width: kWidth, height: kHeight, depth: 1 },
+          mipLevelCount: 1,
+
+          usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.OUTPUT_ATTACHMENT,
+        })
+      );
+    }
+
+    const colorAttachmentViews: GPUTextureView[] = [];
+
+    for (let i = 0; i < t.params.colorAttachments; i++) {
+      colorAttachmentViews.push(colorAttachments[i].createView());
+    }
+
+    // Make an array of storeOps. Use storeOperation1 for even numbered attachments and
+    // storeOperation2 for odd numbered attachments.
+    const storeOps: GPUStoreOp[] = [];
+    for (let i = 0; i < t.params.colorAttachments; i++) {
+      if (i % 2 === 0) {
+        storeOps.push(t.params.storeOperation1);
+      } else {
+        storeOps.push(t.params.storeOperation2);
+      }
+    }
+
+    // Color load operation will clear to {1.0, 0.0, 0.0, 1.0}
+    // Color store operation is determined by test params.
+    const renderPassColorAttachmentDescriptors: GPURenderPassColorAttachmentDescriptor[] = [];
+    for (let i = 0; i < t.params.colorAttachments; i++) {
+      renderPassColorAttachmentDescriptors.push({
+        attachment: colorAttachmentViews[i],
+        loadValue: { r: 1.0, g: 0.0, b: 0.0, a: 1.0 },
+        storeOp: storeOps[i],
+      });
+    }
+
+    const encoder = t.device.createCommandEncoder();
+    const pass = encoder.beginRenderPass({
+      colorAttachments: renderPassColorAttachmentDescriptors,
+    });
+    pass.endPass();
+    t.device.defaultQueue.submit([encoder.finish()]);
+
+    // Check that the correct store operation occurred.
+    let expectedValue: PerTexelComponent<number> = {};
+    for (let i = 0; i < t.params.colorAttachments; i++) {
+      if (storeOps[i] === 'clear') {
+        // If colorStoreOp was clear, the texture should now contain {0.0, 0.0, 0.0, 0.0}.
+        expectedValue = { R: 0.0, G: 0.0, B: 0.0, A: 0.0 };
+      } else if (storeOps[i] === 'store') {
+        // If colorStoreOP was store, the texture should still contain {1.0, 0.0, 0.0, 1.0}.
+        expectedValue = { R: 1.0, G: 0.0, B: 0.0, A: 1.0 };
+      }
+      t.expectSingleColor(colorAttachments[i], kColorFormat, {
+        size: [kHeight, kWidth, 1],
+        exp: expectedValue,
+      });
+    }
+  });
+
+// Tests that render pass depth stencil store operations work correctly for all renderable color
+// formats, mip levels and array layers.
+g.test('render_pass_store_op,depth_stencil_attachment_only')
+  .params(
+    params()
+      .combine(poptions('depthStencilFormat', kTextureFormats))
+      // Filter out color and non-renderable formats.
+      .unless(
+        ({ depthStencilFormat }) =>
+          kTextureFormatInfo[depthStencilFormat].color ||
+          !kTextureFormatInfo[depthStencilFormat].renderable ||
+          !kTextureFormatInfo[depthStencilFormat].copyable
```

Comment:
Hmm I think it would probably be worth generalizing `expectSingleColor` to do this.
We can always copy from the stencil aspect though -- currently unimplemented in Dawn.

Reply:
Actually we could probably go beyond `expectSingleColor` and make a full `expectContents` helper. For the stencil we would copy. For depth32 we would copy, and for depth24 we would draw a bunch of points at the desired depth and then do depth test equals.

Maybe that's more than we need though.

Reply:
I can write another test. Making expectSingleColor more generic to accept depth24 formats seems ideal. Do I need to be the one to do this? - and if so, I guess it needs to be done with WGSL?

Reply:
Could be done with GLSL in the glsl-dependent branch, but not urgent. Let's leave a TODO for this.

Reply:
TODO added.